src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const array = [...sourceString.split(';')];
+
+  array.map((element, i) => [
+    element.slice(0, element.indexOf(':') + 1),
+    element.slice(element.indexOf(':') + 1, -1),
+  ]);
```

Comment:
The `map` function is used here to transform each element of the array into a pair of strings. However, the transformation logic is incorrect because it does not return the transformed elements. You should return the array of key-value pairs from the `map` function.